src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -112,7 +113,7 @@ void TimerExpiredCallback(System::Layer * systemLayer, void * callbackContext)
 {
     EndpointId endpoint = static_cast<EndpointId>(reinterpret_cast<uintptr_t>(callbackContext));
 
-    Delegate * delegate = GetDelegate(endpoint);
+    auto delegate = GetDelegate(endpoint);
```

Comment:
Is there a reason to change these? The delta seems ok (it is obvious that GetDelegate returns a Delegate) however this makes the diff larger as it stands.

Reply:
It was going to have to change anyway, as Thermostat::Delegate collides with FabricTable::Delegate, and seem cleaner to just switch to auto.

Reply:
I was assuming bareword `Delegate` was now ambiguous with the new header include... but I am not seeing that; it seems to work fine.  Why _was_ this change made?

Reply:
Why does it collide with `FabricTable::Delegate`?  Doesn't seem to...

Reply:
Sorry, I missed a subtlety here; I was getting the compile error on the calls to GetDelegate from methods on ThermostatAttrAccess, but I replaced Delegate * with auto everywhere, even in this function which has no conflict as it's outside of the class. I can revert those lines if you want.

Reply:
OK, reverted the lines without a collision.